### PR TITLE
FIX libuv getaddrinfo EAI errors

### PR DESCRIPTION
**Pull request by t4lz**
## Summary
- return `EAI_*` codes from the Unix `getaddrinfo` detour instead of generic hook error conversions
- keep generic hook-error side effects for non-DNS failures, but collapse them to `EAI_FAIL` for `getaddrinfo` callers
- add a Node regression test that reproduces the empty-remote-DNS-response path without aborting in libuv

## Root Cause
`getaddrinfo(3)` reports failures through its return value, not through `errno`. The Unix mirrord detour was using the generic `Detour::Error -> HookError::into()` path, so `HookError::DNSNoName` and similar non-`DnsLookup` errors could become `-1`/`errno`-style failures instead of valid `EAI_*` results. libuv validates those return codes and aborts when it sees an unknown value.

## Fix
Handle Unix `getaddrinfo` errors explicitly. `ResponseError::DnsLookup` still maps to the native DNS codes, `DNSNoName` now maps to `EAI_NONAME`, and other hook errors preserve their logging/graceful-exit behavior before returning `EAI_FAIL` to the caller.

### CODE REVIEW

File: mirrord/layer-lib/src/error.rs [hunk 1]
```diff
@@ -539,6 +539,25 @@ fn translate_dns_fail(dns_fail: DnsLookupError) -> i32 {
...
+/// return value.
+#[cfg(unix)]
+pub fn getaddrinfo_error_code(fail: HookError) -> i32 {
+    match fail {
+        HookError::ResponseError(ResponseError::DnsLookup(dns_fail)) => {
+            translate_dns_fail(dns_fail)
+        }
+        HookError::DNSNoName => libc::EAI_NONAME,
+        other => {
+            let _ = i64::from(other);
```
Comment (iniw) [inline mirrord/layer-lib/src/error.rs:556]: Unused?

File: mirrord/layer-lib/src/error.rs [hunk 2]
```diff
@@ -666,3 +685,37 @@ impl From<HookError> for *mut hostent {
...
+    }
+
+    #[test]
+    fn getaddrinfo_maps_other_hook_errors_to_eai_fail() {
+        assert_eq!(
+            getaddrinfo_error_code(HookError::NullPointer),
+            libc::EAI_FAIL
+        );
+    }
+}
```
Comment (iniw) [inline mirrord/layer-lib/src/error.rs]: I don't see the value in these unit tests since we have the integration test. The function is super tiny anyway.

**Review by 0x00A5 — DISMISSED** [review]

File: mirrord/layer-lib/src/error.rs (hunk 2 above)
Comment (t4lz) [reply mirrord/layer-lib/src/error.rs]: removed UTs

File: mirrord/layer-lib/src/error.rs (hunk 1 above)
Comment (t4lz) [reply mirrord/layer-lib/src/error.rs:556]: This conversion has side effects - graceful exit, logs, etc.

Comment (t4lz) [reply mirrord/layer-lib/src/error.rs:556]: Adding comment to explain

**Review by iniw — APPROVED** [review]
